gbfs-validator/__test__/gbfs.v3.0-RC.test.js

```diff
@@ -0,0 +1,261 @@
+    })
+  })
+})
+
+describe('exaustive feed', () => {
+  let gbfsFeedServer
+
+  beforeAll(async () => {
+    const { MockRequests } = require('./fixtures/v3.0-RC/exaustive')
+
+    let mockRequests = new MockRequests()
+
+    gbfsFeedServer = mockRequests.build()
+
+    await gbfsFeedServer.listen(serverOpts)
+  })
+
+  afterAll(() => {
+    return gbfsFeedServer.close()
+  })
+
+  test('should validate feed', async () => {
+    const url = `http://${gbfsFeedServer.server.address().address}:${
+      gbfsFeedServer.server.address().port
+    }`
+    const gbfs = new GBFS(`${url}/gbfs.json`)
+
+    expect.assertions(2)
+
+    return gbfs.validation().then((result) => {
+      expect(get_errors(result)).toEqual({
+        errors: [],
+        otherErrors: [],
+        warnings: []
+      })
+
+      expect(result).toMatchObject({
+        summary: expect.objectContaining({
+          version: { detected: '3.0-RC', validated: '3.0-RC' },
+          validatorVersion: '1.0.0',
+          hasErrors: false
+        }),
+        files: expect.any(Array)
+      })
+    })
+  })
+})
+
+describe('default_reserve_time REQUIRED if reservation_price_per_min or reservation_price_flat_rate are defined', () => {
+  let gbfsFeedServer
+
+  beforeAll(async () => {
+    const { MockRequests } = require('./fixtures/v3.0-RC/exaustive')
+
+    class InvalidMockRequests extends MockRequests {
+      system_pricing_plans(...args) {
+        const json = super.system_pricing_plans(...args)
+
+        // This plan with require a default_reserve_time on vehicle_types using it.
+        json.data.plans[0].reservation_price_per_min = 1
+        json.data.plans[0].reservation_price_flat_rate = 2
+
+        // This plan will not.
+        delete json.data.plans[1].reservation_price_per_min
+        delete json.data.plans[1].reservation_price_flat_rate
+
+        return json
+      }
+
+      vehicle_types(...args) {
+        const json = super.vehicle_types(...args)
+
+        const system_pricing_plans = super.system_pricing_plans(...args)
+        let id_requiring = system_pricing_plans.data.plans[0].plan_id
+        let id_not_requiring = system_pricing_plans.data.plans[1].plan_id
+
+        // This invalid vehicle_type will require a default_reserve_time.
+        delete json.data.vehicle_types[0].default_reserve_time
+        json.data.vehicle_types[0].default_pricing_plan_id = id_requiring
+        json.data.vehicle_types[0].pricing_plan_ids = [
+          id_requiring,
+          id_not_requiring
+        ]
+
+        // This valid vehicle_type will not.
+        delete json.data.vehicle_types[1].default_reserve_time
+        json.data.vehicle_types[1].default_pricing_plan_id = id_not_requiring
+        json.data.vehicle_types[1].pricing_plan_ids = [id_not_requiring]
+
+        return json
+      }
+    }
+
+    let mockRequests = new InvalidMockRequests()
+
+    gbfsFeedServer = mockRequests.build()
+
+    await gbfsFeedServer.listen()
```

Comment:
Same suggestion as in https://github.com/MobilityData/gbfs-validator/pull/104#discussion_r1234474674. Thank you 

```suggestion
    await gbfsFeedServer.listen(serverOpts)
```

Reply:
(it's fixed now)